### PR TITLE
Task 7 POST /api/reviews/:reviews/comments

### PR DESCRIPTION
**Pull request by jesswillan**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -141,4 +141,37 @@ describe('app', () => {
         });
     });
   });
+  describe.only('POST /api/reviews/:review_id/comments', () => {
```
Comment (robert112233) [inline __tests__/app.test.js]: .only is naughty

File: db/controllers/comments-controller.js [hunk 2]
```diff
@@ -16,3 +17,18 @@ exports.getCommentsByReviewId = (req, res, next) => {
       next(err);
     });
 };
+
+exports.postComment = (req, res, next) => {
+  console.log(req.body, '<<<reqbody')
```
Comment (robert112233) [inline db/controllers/comments-controller.js]: Remove console.log

File: db/controllers/error-handling-controller.js [hunk 3]
```diff
@@ -9,7 +9,9 @@ exports.handlePSQLErrors = (err, req, res, next) => {
   if (err.code === '22P02') {
     res.status(400).send({msg: 'Bad Request'});
   }
-  next(err);
+  if (err.code === '23503') {
+    res.status(400).send({msg: 'Bad Request'})
+  } next(err);
```
Comment (robert112233) [inline db/controllers/error-handling-controller.js:17]: Cool handler 😎

File: db/models/comments-model.js [hunk 4]
```diff
@@ -14,3 +14,15 @@ exports.fetchCommentsByReviewId = (review_id) => {
       return Promise.reject(err);
     });
 };
+
+exports.insertComment = (username, body, review_id) => {
+  return db.query(
```
Comment (robert112233) [inline db/models/comments-model.js:19]: SQL looks good!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -141,4 +141,37 @@ describe('app', () => {
...
+      };
+      return request(app)
+      .post('/api/reviews/10/comments')
+      .send(requestBody)
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe('Bad Request')
+      })
+    });
+  });
```
Comment (robert112233) [inline __tests__/app.test.js:204]: Missing tests: 

Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999
Status 404, username does not exist { username: "jess", body: "validBody" }

**Review by robert112233 — APPROVED** [review]
Great work here Jess! Just a few missing tests (see below)

**Comment by Harrison365** [comment]
Hi Jess,

Currently, your tests on lines 163 and 184 will do the same thing as they are both non-existent usernames. We would like the one with a blank username to not have the key at all (i.e. req.body = {body: "message here"} only).

Apart from that, its good to go. 👍

**Pull request merged**